### DISCUSSION TITLE
Excel Fill Handle function call

### DISCUSSION THREAD

**Discussion by schebotar** [Excel-DNA/ExcelDna#545] [Q&A] [answered]
Hello
I'm playing with simple function returning random double
```csharp

        [ExcelFunction]
        public static double Random()
        {
            return new Random().NextDouble();
        }
```
This works nice on =Random() in cell or Ctrl + C,V cell with function.
But when I try to copy it with fill handle I just get one random number
![image](https://user-images.githubusercontent.com/51533848/209907456-c26af3b7-55d2-4536-af30-2fdfc1fabad3.png)

Is it possible to call function on each row filled with fill handle?

**Comment by govert** [comment, marked as answer]
If you mark your function as volatile with `[ExcelFunction(IsVolatile=true)]` it will recalculate and return a new value every time the sheet calculates, just like the built-in `=RAND()` function.
To get a good distribution of random numbers, you should also create one instance of the .NET 'Random' class, and use that one repeatedly. 

```

       static Random _rand = new Random();

       [ExcelFunction(IsVolatile=true)]
        public static double Random()
        {
            return _rand.NextDouble();
        }
```
